clients/src/main/java/org/apache/kafka/clients/producer/internals/ProducerBatch.java

```diff
@@ -268,6 +268,8 @@ private void completeFutureAndFireCallbacks(long baseOffset, long logAppendTime,
             // A newly created batch can always host the first message.
             if (!batch.tryAppendForSplit(record.timestamp(), record.key(), record.value(), record.headers(), thunk)) {
                 batches.add(batch);
+                // Release compression outputstream.
+                batch.closeForRecordAppends();
```

Comment:
We probably also want to close the last batch for append in line 279.

Reply:
@becketqin My negligence! The last batch should also be closed. I have modified it.